pkg/querier/queryrange/extensions.go

```diff
@@ -30,6 +34,10 @@ func (m *LokiSeriesResponse) GetHeaders() []*queryrangebase.PrometheusResponseHe
 	return nil
 }
 
+func (m *LokiSeriesResponse) SetHeader(name, value string) {
+	m.Headers = append(deleteHeader(m.Headers, name), queryrangebase.PrometheusResponseHeader{Name: name, Values: []string{value}})
```

Comment:
Since this code is repeated across many types, I'd suggest we move it to a common function and test its behaviour

Reply:
Done.